pkg/kubeconfig/kubecfg_builder.go

```diff
@@ -116,25 +117,15 @@ func (b *KubeconfigBuilder) WriteKubecfg() error {
 			cluster = clientcmdapi.NewCluster()
 		}
 		cluster.Server = b.Server
-
-		if b.CACert == nil {
-			// For now, we assume that the cluster has a "real" cert issued by a CA
-			cluster.InsecureSkipTLSVerify = false
-			cluster.CertificateAuthority = ""
-			cluster.CertificateAuthorityData = nil
-		} else {
-			cluster.InsecureSkipTLSVerify = false
-			cluster.CertificateAuthority = ""
-			cluster.CertificateAuthorityData = b.CACert
-		}
+		cluster.CertificateAuthorityData = b.CACert
 
 		if config.Clusters == nil {
 			config.Clusters = make(map[string]*clientcmdapi.Cluster)
 		}
 		config.Clusters[b.Context] = cluster
 	}
 
-	{
+	if b.KubeUser == "admin" {
```

Comment:
Why only write out the client cert if there is a plaintext password?

Reply:
It is a bit of a hack. I changed the check to something matching kops assumptions on user names.